dLux/apertures.py

```diff
@@ -0,0 +1,1430 @@
+from typing import TypeVar
+from dLux.utils import (get_positions_vector, get_pixel_positions)
+from abc import ABC, abstractmethod 
+import equinox as eqx
+import jax.numpy as np
+import jax 
+import functools
+
+jax.config.update("jax_enable_x64", True)
+
+Array = TypeVar("Array")
+Layer = TypeVar("Layer")
+Tensor = TypeVar("Tensor")
+Matrix = TypeVar("Matrix")
+Vector = TypeVar("Vector")
+
+
+__all__ = ["Aperture", "CompoundAperture", "SoftEdgedAperture", 
+    "SquareAperture", "SoftEdgedSquareAperture", "RectangularAperture",
+    "SoftEdgedRectangularAperture", "CircularAperture", 
+    "SoftEdgedCircularAperture", "AnnularAperture", 
+    "SoftEdgedAnnularAperture"]
+
+
+def cartesian_to_polar(coordinates : Tensor) -> Tensor:
```

Comment:
An (almost) identical version of this function already exists in the utils.helpers.py script. Happy to leave this here for now, and I'll remove it when I document the rest of the coordinate transforms